### ISSUE TITLE
`db.destroy`: documented to accept an options parameter, failing when supplied

### ISSUE THREAD

**Issue by thriqon** [pouchdb/pouchdb#4219]
According to the documentation ([api/delete_database](https://github.com/pouchdb/pouchdb/blob/master/docs/_includes/api/delete_database.html)) it is allowed to supply an `options` parameter to the `.destroy` method.

However, the underlying code seems to ignore it and fail, if given ([lib/adapter.js#L793](https://github.com/pouchdb/pouchdb/blob/master/lib/adapter.js#L793))

```javascript
AbstractPouchDB.prototype.destroy =
  utils.adapterFun('destroy', function (callback) {
```
Apparently, no present adapter seems to expect an options parameter in the first place. But since this is documented behavior, I'd suggest changing code to accept it (and maybe ignore it silently?).

**Comment by nolanlawson** [comment]
We actually have quite a few methods where it doesn't take any options. (E.g. `revsDiff()` and `viewCleanup()`.) We only add `options` to the docs because 1) it applies to the http adapter, because you can always pass ajax options in there, and because 2) it's more consistent that way.

In this case, you can indeed set things like `ajax.timeout` for `destroy()` (see `http/index.js`, it overrides the line of code you pointed out).

**Comment by thriqon** [comment]
Unfortunately, that is not the case: http://jsbin.com/wenaba/6/edit?js,output

This specific override has been removed in https://github.com/pouchdb/pouchdb/commit/ef247eced792f374c1632d9b55821c3c3e62e473

**Comment by marten-de-vries** [comment]
This would simplify the implementation of ``pouchdb-wrappers`` too. (Although it currently just hacks around the problem succesfully.)

**Comment by daleharvey** [comment]
Yup, I think its best for consistency and to be future proof to accept the options, I suspect we may want to be able to specify ajax option in this in future although I wont make it part of this patch, will add the options and make sure the call doesnt fail, we can follow up to add and test options

**Comment by nolanlawson** [comment]
fixed by 2995c925fa57f8996be09a337803a5f4b6b3b3a1
